### PR TITLE
Refactor/bac 61 refactor trend endpoint

### PR DESCRIPTION
**Pull request by jayjaytentacion**
add function to query db for rates by limit in backend/app/crud/rate.py
refactor (get) /history/{isocode} endpoint to return n <= 15 rates

sample request: http://127.0.0.1:8000/api/rate/history/NGN
sample response: {
    "status_code": 200,
    "data": {
        "currency": {
            "id": 1,
            "symbol": null,
            "isocode": "NGN",
            "country": "Nigeria"
        },
        "rates": []
    }
}

### CODE REVIEW

File: backend/app/api/v1/rate.py [hunk 1]
```diff
@@ -25,23 +25,28 @@ def getRate(isocode, db: Session = Depends(get_db)):
...
-        }
-    return {"success": True, "status_code": 200, "data": {"currency": currency, "rate": rate}}
+        raise HTTPException(status_code=404, detail="Currency not found")
 
+    #restrict limit to < 15 rate objects
+    if limit > 15:
+        raise HTTPException(status_code=400, detail="Not more than 15 rates per request")
+
+    #get rate objects based on limit set
+    rate = crud.rate.get_rates_by_limit(db,currency_id=currency.id, limit=limit)
```
Comment (ibkay998) [inline backend/app/api/v1/rate.py:42]: Use the get_multi function inside the crud.base and delete the one you created in crud.rate

File: backend/app/crud/rate.py [hunk 2]
```diff
@@ -10,6 +10,15 @@
 
 class CRUDRate(CRUDBase[Rate, RateCreate, RateBase]):
     # Declare model specific CRUD operation methods.
+    def get_rates_by_limit(self, db: Session, currency_id, skip: int = 0, limit: int = 15):
```
Comment (ibkay998) [inline backend/app/crud/rate.py:13]: remove this and use get multi in base same as last review

**Comment by ibkay998** [comment]
close this pull request and make a pull request to dev. dev is our main branch

File: backend/app/api/v1/rate.py (hunk 1 above)
Comment (jayjaytentacion) [reply backend/app/api/v1/rate.py:42]: The get_multi function does not filter the rate based on a currency, it returns all currency objects

**Comment by jayjaytentacion** [comment]
> close this pull request and make a pull request to dev. dev is our main branch

I've changed the base branch to dev

File: backend/app/api/v1/rate.py (hunk 1 above)
Comment (ibkay998) [reply backend/app/api/v1/rate.py:42]: you can specify limit inside the get_multi

**Comment by ibkay998** [comment]
i mentioned you can specify limit inside the get_multi you are yet to implement that this is to ensure we are not duplicating code.

File: backend/app/api/v1/rate.py (hunk 1 above)
Comment (jayjaytentacion) [reply backend/app/api/v1/rate.py:49]: it's not about the limit, the get_multi function doesn't filter rate objects based on a currency

**Review by jpphilips — APPROVED** [review]

**Comment by jayjaytentacion** [comment]
> i mentioned you can specify limit inside the get_multi you are yet to implement that this is to ensure we are not duplicating code.

it's not about the limit, the get_multi function doesn't filter rate objects based on a currency

**Pull request merged**